### PR TITLE
provide alternative rustic mode that derives from rust-ts-mode

### PR DESCRIPTION
**Pull request by brotzeit**
#475

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,6 +136,29 @@ additional [installation
 instructions](https://github.crookster.org/switching-to-straight.el-from-emacs-26-builtin-package.el/)
 for moving your package management from package.el to straight.
 
+## treesitter
+
+You can try the new native treesitter mode `rust-ts-mode` with:
+
+```elisp
+(use-package rustic
+  :init
```
Comment (psibi) [inline README.md:145]: Probably this is more idiomatic:

```
:custom
(rustic-treesitter-derive t)
```

**Review by psibi — COMMENTED** [review]
This looks great!

File: README.md (hunk 1 above)
Comment (brotzeit) [reply README.md:145]: We need to set the variable before rustic.el is required. Unfortunately we can't set the parent mode dynamically(at least as far I know) so we have to configure this at initialization.

Comment (psibi) [reply README.md:146]: Ah, I see. Thanks!

**Comment by brotzeit** [comment]
Ok let's give it a try. Shouldn't break anything, hopefully =)

**Comment by brotzeit** [comment]
I'll revert it for now. Seems I only tested it with emacs master. I'll check again.

**Comment by brotzeit** [comment]
The issue was probably that I tagged the variable with emacs29 afterwards...

**Comment by CeleritasCelery** [comment]
Now that Emacs 29 is official released. Can this be merged again?

**Comment by frtmelody** [comment]
@brotzeit Emacs 29 is official released, treesit should be supported?

**Comment by dertuxmalwieder** [comment]
I'll +1 here. I would like to slowly migrate to `rust-ts-mode` (and others), but Rustic is a great help, so I would like those two to play together well.

**Comment by psibi** [comment]
@liebkne and others: Can you give this a try and see how it goes: https://github.com/brotzeit/rustic/pull/527 ? I have been using it for a day and seems to be working fine.

**Pull request merged**